crates/vsock/src/thread_backend.rs

```diff
@@ -6,22 +6,46 @@ use std::{
         net::UnixStream,
         prelude::{AsRawFd, FromRawFd, RawFd},
     },
+    sync::{Arc, RwLock},
 };
 
 use log::{info, warn};
-use virtio_vsock::packet::VsockPacket;
+use virtio_vsock::packet::{VsockPacket, PKT_HEADER_SIZE};
 use vm_memory::bitmap::BitmapSlice;
 
 use crate::{
     rxops::*,
     vhu_vsock::{
-        ConnMapKey, Error, Result, VSOCK_HOST_CID, VSOCK_OP_REQUEST, VSOCK_OP_RST,
+        CidBkndMap, ConnMapKey, Error, Result, VSOCK_HOST_CID, VSOCK_OP_REQUEST, VSOCK_OP_RST,
         VSOCK_TYPE_STREAM,
     },
     vhu_vsock_thread::VhostUserVsockThread,
     vsock_conn::*,
 };
 
+pub(crate) struct RawVsockPacket {
+    pub header: [u8; PKT_HEADER_SIZE],
+    pub data: Vec<u8>,
+}
+
+impl RawVsockPacket {
+    fn create_from_vsock_packet<B: BitmapSlice>(pkt: &VsockPacket<B>) -> Result<Self> {
```

Comment:
nit: usually, this type of constructors/factory methods are called just `from_` instead of `create_from_`